src/diagnostic_variables.jl

```diff
@@ -202,13 +243,17 @@ function Base.zeros(::Type{DiagnosticVariablesLayer},
     return DiagnosticVariablesLayer(tendencies,grid_variables,dynamics_variables,npoints)
 end
 
+# GPU methods for DiagnosticVariablesLayer
+function Adapt.adapt_structure(to, dvl::DiagnosticVariablesLayer)
+    DiagnosticVariablesLayer(dvl.tendencies, dvl.grid_variables, dvl.dynamics_variables, dvl.npoints)
```

Comment:
Here, the calls to `adapt` are missing for `tendencies`, `grid_variables` and `dynamics_variables`